parser_library/test/workspace/b4g_integration_test.cpp

```diff
@@ -459,3 +471,112 @@ TEST(b4g_integration_test, b4g_conf_noproc_proc_group_default)
+    fm.did_open_file(pgm_a, 1, "");
+    fm.did_open_file(pgm_b, 1, "");
+    fm.did_open_file(pgm_a_diff_path, 1, "");
+
+    workspace_test ws(fm);
+
+    run_if_valid(ws.did_open_file(pgm_a));
+    parse_all_files(ws);
+    ws.collect_diags();
+    EXPECT_TRUE(matches_message_codes(ws.diags(), { "B4G002" }));
+
+    ws.include_non_critical_configuration_diagnostics(true);
+    ws.diags().clear();
+    ws.collect_diags();
+    EXPECT_TRUE(matches_message_codes(ws.diags(), { "B4G002", "B4G003" }));
+
+    run_if_valid(ws.did_close_file(pgm_a));
+    ws.diags().clear();
+    ws.collect_diags();
+    EXPECT_TRUE(ws.diags().empty());
+
+    ws.include_non_critical_configuration_diagnostics(false);
+    ws.diags().clear();
+    ws.collect_diags();
+    EXPECT_TRUE(ws.diags().empty());
+
+    run_if_valid(ws.did_open_file(pgm_b));
+    parse_all_files(ws);
+    ws.diags().clear();
+    ws.collect_diags();
+    EXPECT_TRUE(matches_message_codes(ws.diags(), { "B4G002" }));
+
+    run_if_valid(ws.did_close_file(pgm_b));
+    parse_all_files(ws);
+    ws.diags().clear();
+    ws.collect_diags();
+    EXPECT_TRUE(ws.diags().empty());
+
+    run_if_valid(ws.did_open_file(pgm_a_diff_path));
+    ws.diags().clear();
+    ws.collect_diags();
+    EXPECT_TRUE(ws.diags().empty());
+}
+
+TEST(b4g_integration_test, missing_proc_group_diags_wildcards)
+{
+    file_manager_impl_test fm;
+
+    fm.did_open_file(proc_grps_rl, 1, R"({"pgroups":[{"name":"P1","libs":[]}]})");
+    // Wildcards implicitly in default proc group
+    fm.did_open_file(b4g_conf_rl,
+        1,
+        R"({"elements":{"A":{"processorGroup":"P1"}},"defaultProcessorGroup":"P2","fileExtension":""})");
+    fm.did_open_file(pgm_a, 1, "");
+
+    workspace_test ws(fm);
+
+    run_if_valid(ws.did_open_file(pgm_a));
+    parse_all_files(ws);
+    ws.collect_diags();
+    EXPECT_TRUE(ws.diags().empty());
+
+    ws.include_non_critical_configuration_diagnostics(true);
+    ws.diags().clear();
+    ws.collect_diags();
+    EXPECT_TRUE(matches_message_codes(ws.diags(), { "B4G003" }));
+
+    run_if_valid(ws.did_close_file(pgm_a));
+    ws.diags().clear();
+    ws.collect_diags();
+    EXPECT_TRUE(ws.diags().empty());
+}
+
+TEST(b4g_integration_test, missing_proc_group_diags_wildcards_noproc)
+{
+    file_manager_impl_test fm;
+
+    fm.did_open_file(proc_grps_rl, 1, empty_proc_grps);
+    fm.did_open_file(b4g_conf_rl,
+        1,
+        R"({"elements":{"A":{"processorGroup":"*NOPROC*"}},"defaultProcessorGroup":"P2","fileExtension":""})");
+    fm.did_open_file(pgm_a, 1, "");
+
+    workspace_test ws(fm);
+
+    run_if_valid(ws.did_open_file(pgm_a));
+    parse_all_files(ws);
+    ws.collect_diags();
+    EXPECT_TRUE(ws.diags().empty());
+
+    ws.include_non_critical_configuration_diagnostics(true);
+    ws.diags().clear();
+    ws.collect_diags();
+    EXPECT_TRUE(matches_message_codes(ws.diags(), { "B4G003" }));
```

Comment:
This is for the `P2` default group, right?

Reply:
Yup, that's right

Reply:
That's ok... I just wanted to make sure I understood the test.